pkg/network/protocols/http/model_windows.go

```diff
@@ -106,14 +106,37 @@ func (tx *WinHttpTransaction) StaticTags() uint64 {
 //
 //nolint:revive // TODO(WKIT) Fix revive linter
 func (tx *WinHttpTransaction) DynamicTags() []string {
-	if len(tx.AppPool) != 0 || len(tx.SiteName) != 0 {
-		return []string{
-			fmt.Sprintf("http.iis.app_pool:%v", tx.AppPool),
-			fmt.Sprintf("http.iis.site:%v", tx.SiteID),
-			fmt.Sprintf("http.iis.sitename:%v", tx.SiteName),
-		}
+	tags := make([]string, 0, 6)
+	tags = append(tags, fmt.Sprintf("http.iis.site:%v", tx.SiteID))
+	if (len(tx.AppPool)) > 0 {
+		tags = append(tags, fmt.Sprintf("http.iis.app_pool:%v", tx.AppPool))
 	}
-	return nil
+	if (len(tx.SiteName)) > 0 {
+		tags = append(tags, fmt.Sprintf("http.iis.sitename:%v", tx.SiteName))
+	}
+
+	// tag precedence is web.config -> datadog.json
+	if (len(tx.TagsFromConfig.DDEnv)) > 0 {
+		tags = append(tags, fmt.Sprintf("env:%v", tx.TagsFromConfig.DDEnv))
+	} else if (len(tx.TagsFromJson.DDEnv)) > 0 {
+		tags = append(tags, fmt.Sprintf("env:%v", tx.TagsFromJson.DDEnv))
+	}
+
+	if (len(tx.TagsFromConfig.DDService)) > 0 {
+		tags = append(tags, fmt.Sprintf("service:%v", tx.TagsFromConfig.DDService))
+	} else if (len(tx.TagsFromJson.DDService)) > 0 {
+		tags = append(tags, fmt.Sprintf("service:%v", tx.TagsFromJson.DDService))
+	}
+
+	if (len(tx.TagsFromConfig.DDVersion)) > 0 {
+		tags = append(tags, fmt.Sprintf("version:%v", tx.TagsFromConfig.DDVersion))
+	} else if (len(tx.TagsFromJson.DDVersion)) > 0 {
+		tags = append(tags, fmt.Sprintf("version:%v", tx.TagsFromJson.DDVersion))
+	}
+	if len(tags) == 0 {
```

Comment:
This is dead code this tags is guaranteed to at least have: fmt.Sprintf("http.iis.site:%v", tx.SiteID)

Reply:
good catch.  I accidentally changed the behavior above when fixing another subtle request.
Code is no longer dead.